### PR TITLE
gcc7: cmn_err() supports 'h' and 'hh'

### PR DESCRIPTION
**Pull request by citrus-it**

### CODE REVIEW

File: gcc/config/sol2-c.c [hunk 1]
```diff
@@ -60,14 +62,15 @@ static const format_char_info bitfield_string_type =
 
 static const format_char_info cmn_err_char_table[] =
 {
+  /*                     none     hh       h        l        ll       L        z        t        j        H       D       DD */
   /* C89 conversion specifiers.  */
```
Comment (richlowe) [inline gcc/config/sol2-c.c:66]: Did this table go wrong sometime before, or am I missing why it grows 3 extra feels for two extra formats?
(I'm guessing the latter?)

File: gcc/testsuite/gcc.dg/format/cmn-err-1.c [hunk 2]
```diff
@@ -26,6 +26,8 @@ int main()
   cmn_err_func (0, "%b %s", i, "\01Foo", string);
   cmn_err_func (0, "%p", string);
   cmn_err_func (0, "%16b", i, "\01Foo");
+  cmn_err_func (0, "%hu", 0xabcd);
```
Comment (richlowe) [inline gcc/testsuite/gcc.dg/format/cmn-err-1.c]: Is there any reason to test with other modifiers?

File: gcc/config/sol2-c.c (hunk 1 above)
Comment (citrus-it) [reply gcc/config/sol2-c.c:66]: It was not updated when the commit below went into gcc, growing the list of formats. BADLEN is `{0, NULL, NULL}` so it did not matter, but I expanded it to match while I was in here.

```
commit 9a8ce21f3b47a109b9ac35239e21198bb36e1146
Author: Jon Grimm <jgrimm2@us.ibm.com>
Date:   Fri Dec 16 06:38:26 2005 +0000
            * c-format.h (enum format_lengths): Add new enumeration values
            FMT_LEN_H, FMT_LEN_D and FMT_LEN_DD.
            * c-format.c (printf_length_specs, scanf_length_specs): Add
            entries for H, D, DD.
```

File: gcc/testsuite/gcc.dg/format/cmn-err-1.c (hunk 2 above)
Comment (citrus-it) [reply gcc/testsuite/gcc.dg/format/cmn-err-1.c]: I just hadn't pushed the latest branch.

**Comment by citrus-it** [comment]
Adding testsuite results for the various variations:

```
10:Running target unix//-m32
212191:Executing on host: /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/  /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c  -m32   -fno-diagnostics-show-caret -fdiagnostics-color=never     -Wformat -S -o cmn-err-1.s    (timeout = 300)
212192:spawn -ignore SIGHUP /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/ /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c -m32 -fno-diagnostics-show-caret -fdiagnostics-color=never -Wformat -S -o cmn-err-1.s
212193:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c: In function 'main':
212194:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:22: warning: unknown conversion type character 'i' in format [-Wformat=]
212195:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:20: warning: too many arguments for format [-Wformat-extra-args]
212196:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:36:22: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
212197:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'int' argument [-Wformat=]
212198:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
212199:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:39:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
212200:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:40:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
212201:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'int', but argument 3 has type 'const char *' [-Wformat=]
212202:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
212203:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:42:22: warning: format '%p' expects argument of type 'void *', but argument 3 has type 'int' [-Wformat=]
212204:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 35)
212205:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 36)
212206:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 37)
212207:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 37)
212208:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 39)
212209:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 40)
212210:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 41)
212211:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 42)
212212:PASS: gcc.dg/format/cmn-err-1.c    (test for excess errors)
212213:Executing on host: /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/  /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c  -m32   -fno-diagnostics-show-caret -fdiagnostics-color=never    -DWIDE  -Wformat -S -o cmn-err-1.s    (timeout = 300)
212214:spawn -ignore SIGHUP /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/ /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c -m32 -fno-diagnostics-show-caret -fdiagnostics-color=never -DWIDE -Wformat -S -o cmn-err-1.s
212215:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c: In function 'main':
212216:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:22: warning: unknown conversion type character 'i' in format [-Wformat=]
212217:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:20: warning: too many arguments for format [-Wformat-extra-args]
212218:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:36:22: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
212219:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'int' argument [-Wformat=]
212220:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
212221:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:39:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
212222:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:40:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
212223:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'int', but argument 3 has type 'const char *' [-Wformat=]
212224:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
212225:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:42:22: warning: format '%p' expects argument of type 'void *', but argument 3 has type 'int' [-Wformat=]
212226:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 35)
212227:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 36)
212228:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 37)
212229:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 37)
212230:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 39)
212231:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 40)
212232:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 41)
212233:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 42)
212234:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE  (test for excess errors)
424695:Running target unix//-m64
638319:Executing on host: /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/  /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c  -m64    -fno-diagnostics-show-caret -fdiagnostics-color=never     -Wformat -S -o cmn-err-1.s    (timeout = 300)
638320:spawn -ignore SIGHUP /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/ /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c -m64 -fno-diagnostics-show-caret -fdiagnostics-color=never -Wformat -S -o cmn-err-1.s
638321:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c: In function 'main':
638322:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:22: warning: unknown conversion type character 'i' in format [-Wformat=]
638323:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:20: warning: too many arguments for format [-Wformat-extra-args]
638324:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:36:22: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
638325:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'int' argument [-Wformat=]
638326:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
638327:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:39:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
638328:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:40:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
638329:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'int', but argument 3 has type 'const char *' [-Wformat=]
638330:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
638331:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:42:22: warning: format '%p' expects argument of type 'void *', but argument 3 has type 'int' [-Wformat=]
638332:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 35)
638333:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 36)
638334:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 37)
638335:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 37)
638336:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 39)
638337:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 40)
638338:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 41)
638339:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 42)
638340:PASS: gcc.dg/format/cmn-err-1.c    (test for excess errors)
638341:Executing on host: /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/  /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c  -m64    -fno-diagnostics-show-caret -fdiagnostics-color=never    -DWIDE  -Wformat -S -o cmn-err-1.s    (timeout = 300)
638342:spawn -ignore SIGHUP /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/ /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c -m64 -fno-diagnostics-show-caret -fdiagnostics-color=never -DWIDE -Wformat -S -o cmn-err-1.s
638343:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c: In function 'main':
638344:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:22: warning: unknown conversion type character 'i' in format [-Wformat=]
638345:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:20: warning: too many arguments for format [-Wformat-extra-args]
638346:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:36:22: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
638347:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'int' argument [-Wformat=]
638348:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
638349:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:39:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
638350:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:40:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
638351:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'int', but argument 3 has type 'const char *' [-Wformat=]
638352:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
638353:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:42:22: warning: format '%p' expects argument of type 'void *', but argument 3 has type 'int' [-Wformat=]
638354:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 35)
638355:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 36)
638356:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 37)
638357:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 37)
638358:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 39)
638359:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 40)
638360:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 41)
638361:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 42)
638362:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE  (test for excess errors)
857233:Running target unix//-m64/-msave-args
1070880:Executing on host: /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/  /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c  -m64 -msave-args     -fno-diagnostics-show-caret -fdiagnostics-color=never     -Wformat -S -o cmn-err-1.s    (timeout = 300)
1070881:spawn -ignore SIGHUP /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/ /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c -m64 -msave-args -fno-diagnostics-show-caret -fdiagnostics-color=never -Wformat -S -o cmn-err-1.s
1070882:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c: In function 'main':
1070883:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:22: warning: unknown conversion type character 'i' in format [-Wformat=]
1070884:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:20: warning: too many arguments for format [-Wformat-extra-args]
1070885:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:36:22: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
1070886:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'int' argument [-Wformat=]
1070887:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
1070888:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:39:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
1070889:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:40:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
1070890:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'int', but argument 3 has type 'const char *' [-Wformat=]
1070891:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
1070892:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:42:22: warning: format '%p' expects argument of type 'void *', but argument 3 has type 'int' [-Wformat=]
1070893:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 35)
1070894:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 36)
1070895:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 37)
1070896:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 37)
1070897:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 39)
1070898:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 40)
1070899:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 41)
1070900:PASS: gcc.dg/format/cmn-err-1.c     (test for warnings, line 42)
1070901:PASS: gcc.dg/format/cmn-err-1.c    (test for excess errors)
1070902:Executing on host: /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/  /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c  -m64 -msave-args     -fno-diagnostics-show-caret -fdiagnostics-color=never    -DWIDE  -Wformat -S -o cmn-err-1.s    (timeout = 300)
1070903:spawn -ignore SIGHUP /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/xgcc -B/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0-build/gcc/ /build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c -m64 -msave-args -fno-diagnostics-show-caret -fdiagnostics-color=never -DWIDE -Wformat -S -o cmn-err-1.s
1070904:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c: In function 'main':
1070905:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:22: warning: unknown conversion type character 'i' in format [-Wformat=]
1070906:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:35:20: warning: too many arguments for format [-Wformat-extra-args]
1070907:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:36:22: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
1070908:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'int' argument [-Wformat=]
1070909:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:37:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
1070910:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:39:22: warning: format '%b' expects a matching 'char *' argument [-Wformat=]
1070911:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:40:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
1070912:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'int', but argument 3 has type 'const char *' [-Wformat=]
1070913:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:41:22: warning: format '%b' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
1070914:/build/af/bloody/_build/gcc-7.5.0/gcc-7.5.0/gcc/testsuite/gcc.dg/format/cmn-err-1.c:42:22: warning: format '%p' expects argument of type 'void *', but argument 3 has type 'int' [-Wformat=]
1070915:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 35)
1070916:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 36)
1070917:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 37)
1070918:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 37)
1070919:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 39)
1070920:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 40)
1070921:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 41)
1070922:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE   (test for warnings, line 42)
1070923:PASS: gcc.dg/format/cmn-err-1.c   -DWIDE  (test for excess errors)
```

**Comment by rmustacc** [comment]
Are the warnings in the test output expected?

**Comment by citrus-it** [comment]
> Are the warnings in the test output expected?

Yes, the tests on those lines deliberately test that warnings are generated.

See

https://github.com/illumos/gcc/pull/48/files#diff-2d2a4ba837e900ec406e71384204330b2158d550a18ea8a7d79bf51f3d0c837eR35-R42

**Comment by rmustacc** [comment]
OK, thanks for that context @citrus-it. I think I'm good with moving forward here. @richlowe would you like to check anything else or are you good?

**Comment by rmustacc** [comment]
@richlowe I assume we're good to go here given the fact that we pulled it into gcc10?

**Comment by richlowe** [comment]
Yes.

**Comment by rmustacc** [comment]
This has been landed and tagged as gcc-7.5.0-il-2.

**Comment by citrus-it** [comment]
Thanks all - now we just need to wait until everyone updates their shadows before we can start using `%h` and `%hh` in uts!

**Pull request merged**